design-system/pkg/src/overlays/Overlay.tsx

```diff
@@ -1,10 +1,17 @@
-import { ForwardedRef, forwardRef, useCallback, useState } from 'react';
+import {
+  ForwardedRef,
+  forwardRef,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
 import { Overlay as ReactAriaOverlay } from '@react-aria/overlays';
 import { OverlayProps } from '@react-types/overlays';
 
 import { KeystarProvider } from '@keystar/ui/core';
+import { cloneValidElement } from '../utils';
 
-import { OpenTransition } from './OpenTransition';
+const forceReflow = (node: HTMLElement) => node.scrollTop;
```

Comment:
Won't this just return a number? Not sure how this works, sorry.

If I understand the intent, it seems like idiomatic React would be to wrap the appropriate `setState()` call with [forceUpdate()](https://react.dev/reference/react/Component#forceupdate).

Reply:
Yeah, it triggers a side-effect, it's the same as what react-transition-group does: https://github.com/reactjs/react-transition-group/blob/2989b5b87b4b4d1001f21c8efa503049ffb4fe8d/src/utils/reflow.js